chromecastize.sh

```diff
@@ -30,137 +30,137 @@ DEFAULT_GFORMAT=mkv
-                OUTPUT_VCODEC="$DEFAULT_VCODEC"
-        fi
-        echo "- video: $INPUT_VCODEC -> $OUTPUT_VCODEC"
-
-        # test audio codec
-        INPUT_ACODEC=`mediainfo --Inform="Audio;%Format%\n" "$FILENAME" | head -n1`
-        if is_supported_acodec "$INPUT_ACODEC"; then
-                OUTPUT_ACODEC="copy"
-        else
-                OUTPUT_ACODEC="$DEFAULT_ACODEC"
-        fi
-        echo "- audio: $INPUT_ACODEC -> $OUTPUT_ACODEC"
-
-        if [ "$OUTPUT_VCODEC" = "copy" ] && [ "$OUTPUT_ACODEC" = "copy" ] && [ "$OUTPUT_GFORMAT" = "ok" ]; then
-                echo "- file should be playable by Chromecast!"
-		mark_as_good "$FILENAME"
-	else
-		echo "- video length: `mediainfo --Inform="General;%Duration/String3%" "$FILENAME"`"
-		$FFMPEG -loglevel error -stats -i "$FILENAME" -map 0 -scodec copy -vcodec "$OUTPUT_VCODEC" -acodec "$OUTPUT_ACODEC" "$FILENAME.$OUTPUT_GFORMAT" && on_success "$FILENAME" || on_failure "$FILENAME"
-		echo ""
-        fi
+  echo "==========="
+  echo "Processing: $1"
+
+  # test extension
+  local BASENAME=$(basename "$1")
+  local EXTENSION="${BASENAME##*.}"
+  if ! is_supported_ext "$EXTENSION"; then
+    echo "- not a video format, skipping"
+    continue
+  fi
+
+  # test if it's an `chromecastize` generated file
+  if grep -Fxq "`$REALPATH "$1"`" $HOME/processed_files; then
+    echo '- file was genereated by `chromecastize`, skipping'
+    continue
+  fi
+
+  # test general format
+  local INPUT_GFORMAT=`mediainfo --Inform="General;%Format%\n" "$1" | head -n1`
+  local OUTPUT_GFORMAT=""
+  if is_supported_gformat "$INPUT_GFORMAT" && [ "$OVERRIDE_GFORMAT" = "" ] || [ "$OVERRIDE_GFORMAT" = "$EXTENSION" ]; then
+    OUTPUT_GFORMAT="ok"
+  else
+    # if override format is specified, use it; otherwise fall back to default format
+    OUTPUT_GFORMAT="${OVERRIDE_GFORMAT:-$DEFAULT_GFORMAT}"
+  fi
+  echo "- general: $INPUT_GFORMAT -> $OUTPUT_GFORMAT"
+
+  # test video codec
+  local INPUT_VCODEC=`mediainfo --Inform="Video;%Format%\n" "$1" | head -n1`
+  local OUTPUT_VCODEC=""
+  if is_supported_vcodec "$INPUT_VCODEC"; then
+    OUTPUT_VCODEC="copy"
+  else
+    OUTPUT_VCODEC="$DEFAULT_VCODEC"
+  fi
+  echo "- video: $INPUT_VCODEC -> $OUTPUT_VCODEC"
+
+  # test audio codec
+  local INPUT_ACODEC=`mediainfo --Inform="Audio;%Format%\n" "$1" | head -n1`
+  local OUTPUT_ACODEC=""
+  if is_supported_acodec "$INPUT_ACODEC"; then
+    OUTPUT_ACODEC="copy"
+  else
+    OUTPUT_ACODEC="$DEFAULT_ACODEC"
+  fi
+  echo "- audio: $INPUT_ACODEC -> $OUTPUT_ACODEC"
+
+  if [ "$OUTPUT_VCODEC" = "copy" ] && [ "$OUTPUT_ACODEC" = "copy" ] && [ "$OUTPUT_GFORMAT" = "ok" ]; then
+    echo "- file should be playable by Chromecast!"
+    mark_as_good "$1"
+  else
+    echo "- video length: `mediainfo --Inform="General;%Duration/String3%" "$1"`"
+    $FFMPEG -loglevel error -stats -i "$1" -map 0 -scodec copy -vcodec "$OUTPUT_VCODEC" -acodec "$OUTPUT_ACODEC" "$1.$OUTPUT_GFORMAT" </dev/null && on_success "$1" || on_failure "$1"
```

Comment:
Here I forgot one of the $FILENAME references in the old patch

Also, I added the </dev/null stdin redirect because with my change to read the filenames in the outer loop from stdin, ffmpeg was inheriting that stdin